app/controllers/schools/ects_controller.rb

```diff
@@ -3,7 +3,8 @@ class ECTsController < SchoolsController
     layout "full"
 
     def index
-      @pagy, @teachers = pagy_array(Teachers::Search.new(ect_at_school: school).search)
+      @teachers = Teachers::Search.new(ect_at_school: school).search
+      @pagy, @filtered_teachers = pagy_array(Teachers::Search.new(ect_at_school: school, query_string: params[:q]).search)
```

Comment:
I think we should be able to get away with firing one search here, omitting the query string if it's blank.